plugins/guests/tinycore/cap/change_host_name.rb

```diff
@@ -4,7 +4,6 @@ module Cap
       class ChangeHostName
         def self.change_host_name(machine, name)
           if !machine.communicate.test("hostname | grep '^#{name}$'")
-            machine.communicate.sudo("sh -c 'echo \"#{name}\" > /etc/hostname'")
```

Comment:
@bmx0r on most linux systems, setting the contents of `/etc/hostname` is how you persist the hostname on reboots. I think Vagrant makes the promise to persist that hostname, which is why I'm concerned we are removing that persistence.